### PR TITLE
Add markers and info windows to Map, enable night/day mode selection

### PR DESCRIPTION
**Pull request by lilian-hunt**

### CODE REVIEW

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -6,7 +6,6 @@
     <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css" integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous">
     <link rel="stylesheet" href="style.css">
     <script src="script.js"></script>
-    <script src="map.js"></script>
   </head>
   <body onload="createMap();">
   <div id="content">
```
Comment (aannie-y) [inline portfolio/src/main/webapp/index.html]: should this be indented inside <body>?

File: portfolio/src/main/webapp/map.js [hunk 2]
```diff
@@ -1,6 +1,150 @@
...
+    createMap();
+    button.innerText = 'Night Mode';
+  }
+}
+/** Creates a day mode map and adds it to the page. */
 function createMap() {
   const map = new google.maps.Map(
-      document.getElementById('map'),
-      {center: {lat: 37.422, lng: -122.084}, zoom: 16});
+      document.getElementById('map'), {center: jacksonHole, zoom: 4});
```
Comment (aannie-y) [inline portfolio/src/main/webapp/map.js:59]: 2 space idnentation

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -144,6 +144,8 @@ fetch('./config.json')
       var script = document.createElement('script');
```
Comment (aannie-y) [inline portfolio/src/main/webapp/script.js]: if you can, try avoid using var as per [this](https://google.github.io/styleguide/jsguide.html#features-use-const-and-let)

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -144,6 +144,8 @@ fetch('./config.json')
...
       script.src =
           'https://maps.googleapis.com/maps/api/js?key=' + data.api_key;
-      script.defer = false;
-      document.head.insertBefore(script, document.head.lastChild);
+      document.head.appendChild(script);
+      var mapScript = document.createElement('script');
```
Comment (aannie-y) [inline portfolio/src/main/webapp/script.js]: same here with the 'var'.

**Review by aannie-y — COMMENTED** [review]
Some minor changes here and there regarding indentation and variable declaration but everything else lgtm!

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (lilian-hunt) [reply portfolio/src/main/webapp/script.js]: Thanks for pointing this out. Now replaced all with const

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (lilian-hunt) [reply portfolio/src/main/webapp/script.js]: Replaced with const, as above

File: portfolio/src/main/webapp/index.html (hunk 1 above)
Comment (lilian-hunt) [reply portfolio/src/main/webapp/index.html]: Yes, child elements should be indented.

File: portfolio/src/main/webapp/map.js (hunk 2 above)
Comment (lilian-hunt) [reply portfolio/src/main/webapp/map.js:59]: This is as per clang-format-path .

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -64,23 +64,23 @@ function createListElement(key, text) {
...
 
   // Create an input element to delete item.
-  var form = document.createElement('form');
+  const form = document.createElement('form');
   form.setAttribute('method', 'post');
   form.setAttribute('action', '/delete-data');
 
   // Create hidden element to store id of button.
-  var id = document.createElement('input');
+  const id = document.createElement('input');
```
Comment (jupi3200) [inline portfolio/src/main/webapp/script.js]: id is a very generic term, what is this id supposed to identify?

**Review by jupi3200 — COMMENTED** [review]
This PR contains multiple features (adding markers, night/day mode). It should have been multiple PRs.

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (lilian-hunt) [reply portfolio/src/main/webapp/script.js]: replaced with comment_id, (it's the unique key from the database used to identify the comment)

**Review by KathleenQ — COMMENTED** [review]
The codes look good to me. (I mainly check the format, while also checking the general logic.)

File: portfolio/src/main/webapp/index.html (hunk 1 above)
Comment (tripd22) [reply portfolio/src/main/webapp/index.html]: Has this been fixed? I.e. the indenting of all lines within the \<body> tags

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -97,25 +98,25 @@ function createListElement(key, text) {
...
+      const text = document.createElement('p');
       text.innerText = 'Enter your feedback here:';
 
-      var textArea = document.createElement('textarea');
+      const textArea = document.createElement('textarea');
       textArea.name = 'text-input';
       textArea.required = '';
 
-      var br = document.createElement('br');
+      const br = document.createElement('br');
```
Comment (jupi3200) [inline portfolio/src/main/webapp/script.js]: The name _br_ does not help me understand why we need this.

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -97,25 +98,25 @@ function createListElement(key, text) {
...
   if (this.readyState == 4 && this.status == 200) {
     commentTitle = document.getElementById('comment-header');
     if (xhttp.responseText.includes('<p>You are logged in!</p>')) {
-      var form = document.createElement('form');
+      const form = document.createElement('form');
       form.setAttribute('method', 'POST');
       form.setAttribute('action', '/data');
 
-      var text = document.createElement('p');
+      const text = document.createElement('p');
```
Comment (jupi3200) [inline portfolio/src/main/webapp/script.js]: feedback_prompt?

Comment (lilian-hunt) [reply portfolio/src/main/webapp/script.js]: renamed to feedbackPrompt to be more descriptive

File: portfolio/src/main/webapp/script.js (hunk 6 above)
Comment (lilian-hunt) [reply portfolio/src/main/webapp/script.js]: renamed to breakElement

File: portfolio/src/main/webapp/index.html (hunk 1 above)
Comment (lilian-hunt) [reply portfolio/src/main/webapp/index.html]: Now fixed all the indentation with an online html editor

**Review by jupi3200 — APPROVED** [review]

**Pull request merged**